Makefile

```diff
@@ -17,6 +17,16 @@ build:
 generate:
 	hack/update-generated-bindata.sh
 
+# Generate ClusterIngress CRD from vendored API spec.
```

Comment:
How do we ensure in CI that the generated validations are in sync w/ the vendored types?

Reply:
`make verify-crd` will return `1` if the vendored types differ from the generated validations and vice versa. For example, I added:
```
FooSelector *metav1.LabelSelector `json:"fooSelector,omitempty"`
```
to `NodePlacement` in the vendored `IngressController` type and ran `make verify-crd`:
```
<SNIP>
@@ -99,6 +99,8 @@
                 of the ingress controller.  If unset, defaults are used. See NodePlacement
                 for more details.
               properties:
+                fooSelector:
+                  type: object
                 nodeSelector:
                   description: 'nodeSelector is the node selector applied to ingress
                     controller deployments.  If unset, the default is:    beta.kubernetes.io/os:

verification failed
exit status 1
make: *** [verify-crd] Error 1
```

Should I add an inline doc to the openshift/api section of `Gopkg.toml` saying to run `make verify-crd` after any updates to the dep?

Reply:
If you add it to the `verify` target, CI will catch it during unit test execution (which depends on `verify`).

What about adding the generation step in the existing `generate` target and adding the verification to `verify`?